### PR TITLE
feat: add Stakewise v2 / tweak univ3 / minor fixes

### PR DESCRIPTION
**Pull request by nledo**

### CODE REVIEW

File: sdk/src/protocols/index.ts
Comment (nledo) [inline sdk/src/protocols/index.ts:1]: I'm getting an error here... It seems there's an await missing or sth I can't figure out

Comment (jfschwarz) [reply sdk/src/protocols/index.ts:1]: I don't get an error, looks good to me.

**Review by jfschwarz — APPROVED** [review]

File: app/pages/learn/protocols/stakewise-v2.mdx [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+  | `params.amount1Max` | -                                                     |
+
+<Callout type="info">
+  In the Uniswap v3 PositionsNFT contract, the `collect` function has a special
+  modifier called `isAuthorizedForToken`. This modifier acts like a gatekeeper,
+  ensuring that only authorized senders can perform this action. Uniswap v3 has
+  these checks built into its system, so there's no need for additional
+  constraints.
+</Callout>
+
```
Comment (nledo) [inline app/pages/learn/protocols/stakewise-v2.mdx:177]: It's ok to mention this again, or should we remove it?

Comment (nledo) [reply app/pages/learn/protocols/stakewise-v2.mdx:177]: Maybe we should also add this warning to stakewise:
<Callout type="warning">
  If the avatar has approved the Uniswap v3 Positions NFT contract to spend
  tokens other than the ones specified in `tokens`, the role will be able to
  increase any existing positions' liquidity in these tokens as well.
</Callout>

File: sdk/src/protocols/stakewise/v2/index.ts [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+}
+
+export const eth = {
+  deposit: async ({
+    targets
+  }: {
+    targets: (EthPool["name"] | EthPool["address"])[]
+  }) => {
+    const univ3Permissions = await Promise.all(
+      targets.flatMap((target) =>
```
Comment (jfschwarz) [inline sdk/src/protocols/stakewise/v2/index.ts:30]:
```suggestion
      targets.map((target) =>
```

The array is generally flat list of promises, so `flatMap` has no extra effect over simple `map`

File: sdk/src/protocols/uniswap_v3/test_utils.ts
Comment (jfschwarz) [inline sdk/src/protocols/uniswap_v3/test_utils.ts:1]: file naming convention is actually camelCase, so let's rename to `testUtils.ts`

on that note: shall we rename the folder `uniswap_v3` to `uniswap/v3` to make it consistent with stakewise, compound, aave?

File: app/pages/learn/protocols/stakewise-v2.mdx (hunk 1 above)
Comment (jfschwarz) [reply app/pages/learn/protocols/stakewise-v2.mdx:177]: > It's ok to mention this again, or should we remove it? Should we add these comments in univ3?

I'd say it's not really necessary to mention it. Should be obvious.. If Uniswap would not enforce these constraints anyone could exploit this.

Comment (jfschwarz) [reply app/pages/learn/protocols/stakewise-v2.mdx:177]: adding the warning about open approvals makes total sense

File: app/pages/learn/protocols/uniswap-v3.mdx [hunk 3]
```diff
@@ -147,7 +152,7 @@ Collect accumulated fees from a specific NFT position.
   | `params.amount0Max` | -                 |
   | `params.amount1Max` | -                 |
 
-### If WETH is involved
+### If WETH is involved, the following functions are also whitelisted:
```
Comment (jfschwarz) [inline app/pages/learn/protocols/uniswap-v3.mdx:155]: The longer title is a bit ugly to look at in the nav:

image

I'd vote for the short form, but you can make the call 👍

**Pull request merged**